ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
Oh! From the `Makefile` created by `ruby extconf.rb` (this is what Rake's compiler-plugin will use too):

    warnflags = -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wunused-variable -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat  

and then `$(warnflags)` is included in `$(cflags)` which is included in `$(CFLAGS)`, prior to the flags that we add on.

So we successfully set C99 mode, but `mkmf` already enabled extra warnings (which on gcc apparently warn even for legal C99 constructs), so we need to manually disable those warnings.